google-http-client/src/main/java/com/google/api/client/http/MultipartContent.java

```diff
@@ -47,10 +47,14 @@
   private static final String TWO_DASHES = "--";
 
   /** Parts of the HTTP multipart request. */
-  private ArrayList<Part> parts = new ArrayList<Part>();
+  private ArrayList<Part> parts = new ArrayList<>();
 
   public MultipartContent() {
-    super(new HttpMediaType("multipart/related").setParameter("boundary", "__END_OF_PART__"));
+    this("__END_OF_PART__" + java.util.UUID.randomUUID().toString() + "__");
```

Comment:
Is there a reason not to import java.util.UUID? e.g a conflict with a similarly named class?

Reply:
fixed.